File: app/src/androidTest/java/com/acv/gym/di/EspressoDaggerMockRule.java
package com.acv.gym.di;

import android.support.test.InstrumentationRegistry;

import com.acv.gym.GymApplication;
import com.acv.gym.di.component.AppComponent;
import com.acv.gym.di.module.AppModule;

import it.cosenonjaviste.daggermock.DaggerMockRule;


public class EspressoDaggerMockRule extends DaggerMockRule<AppComponent> {

    public EspressoDaggerMockRule() {
        super(AppComponent.class, new AppModule(getApp()));

        set(new ComponentSetter<AppComponent>() {
            @Override public void setComponent(AppComponent component) {
                GymApplication.appComponent = component;
            }
        });
    }

    private static GymApplication getApp() {
        return (GymApplication) InstrumentationRegistry.getInstrumentation().getTargetContext().getApplicationContext();
    }
}

File: app/src/androidTest/java/com/acv/gym/module/session/EspressoDaggerMockRule.kt
package com.acv.gym.module.session


import com.acv.gym.GymApplication
import com.acv.gym.di.component.AppComponent
import com.acv.gym.di.module.AppModule
import it.cosenonjaviste.daggermock.DaggerMockRule

class EspressoDaggerMockRule(app: GymApplication) : DaggerMockRule<AppComponent>(AppComponent::class.java, AppModule(app)) {
    init {

        set { component -> GymApplication.appComponent = component }
    }

}
